src/tennissimulator/match/Set.java

```diff
@@ -0,0 +1,111 @@
+package tennissimulator.match;
+
+import tennissimulator.player.Player;
+
+import java.util.Optional;
+
+public class Set {
+
+    private static final int GAMES_TO_WIN_SET = 3;
+
+    private final Player firstPlayer;
+    private final Player secondPlayer;
+    private final ScoreTable scoreTable;
+    private final Game game;
+    private Integer firstPlayerGamesWon;
+    private Integer secondPlayerGamesWon;
+
+    public Set(Player firstPlayer, Player secondPlayer, ScoreTable scoreTable) {
+        this.firstPlayer = firstPlayer;
+        this.secondPlayer = secondPlayer;
+        this.scoreTable = scoreTable;
+        game = new Game(firstPlayer, secondPlayer, scoreTable);
+        firstPlayerGamesWon = 0;
+        secondPlayerGamesWon = 0;
+    }
+
+    public Optional<Player> simulate() {
+        while (hasNoWinner()) {
+            updateResult(game.simulate());
+            scoreTable.updateWithGame(firstPlayerGamesWon, secondPlayerGamesWon);
+            changeServer();
+        }
+
+        return getWinner();
+    }
+
+    private void changeServer() {
+        firstPlayer.changeServer();
+        secondPlayer.changeServer();
+    }
+
+    public void resetScore() {
+        firstPlayerGamesWon = 0;
+        secondPlayerGamesWon = 0;
+    }
+
+    private boolean firstPlayerHasWonSet() {
+        return (firstPlayerGamesWon >= GAMES_TO_WIN_SET && firstPlayerGamesWon - secondPlayerGamesWon > 1)
```

Comment:
Why not introducing method from this condition: firstPlayerGamesWon - secondPlayerGamesWon > 1 -> isMultiplePointsDifference()
This method could even be reused easily few lines below

Reply:
created new private method `isMultiplePointsDIfference()`.